### PR TITLE
New login page

### PR DESCRIPTION
**Pull request by nicodeur**
Hello,

I propose a new design for login Page. This page is responsible design.

I used Bootstrap according to cerberus graphical charter.

I tried with a new logo to visualize the result. It will be easy to replace it by the new logo to be choosen by the community.

You can see bellow screenshoot to visualize the new page.

On Desktop: 
login-dekstop
forgotpassword-dekstop

On Mobile : 
login-mobile
forgotpassword-mobile

Regards,

Nicolas

### CODE REVIEW

**Comment by vertigo17** [comment]
I am impressed Nico...
I loged in recently on a mobile device and could feal the pain.
Can we do a version with the today logo that we would replace by the new one once we have chosen.
Thanks a lot.

File: source/src/main/java/org/cerberus/database/DatabaseVersioningService.java [hunk 1]
```diff
@@ -8799,6 +8799,12 @@ public boolean isDatabaseUptodate() {
         SQLS.append("INSERT INTO `invariant` (`idname`, `value`, `sort`, `description`, `VeryShortDesc`) VALUES ('USERGROUP', 'TestStepLibrary', '115', 'Can modify Step Library and flag Step as Library.', '');");
         SQLInstruction.add(SQLS.toString());
 
+        // New design Login page
+        //-- ------------------------ 1158
+        SQLS = new StringBuilder();
+        SQLS.append("UPDATE `parameter` SET value='<a href=\"mailto:support@domain.com?Subject=Cerberus%20Account\">Support</a>' where param='cerberus_support_email'");
```
Comment (vertigo17) [inline source/src/main/java/org/cerberus/database/DatabaseVersioningService.java]: We cannot change a public parameter like that.
That will change the potencial parameter configured by the users.
What can be done is to change the first inserted value on the database so that any new fresh installation of cerberus will have that new value defined. We need to secure though that the original SQL is modified without adding or removing any new SQL.

**Comment by nicodeur** [comment]
Hello @vertigo17 

I agree on the parameter. But I'm a little loosed about it. Html Sentence is actually hardcode on database with an  html/css color 
```
<a href=\"mailto:support@domain.com?Subject=Cerberus%20Account\" style=\"color: yellow\">Support</a>'
```

Color "yellow" should not be hardcoded here. I have some others solutions for it : 
* Indicate on release note the changment, and indicate the database request to launch to resolve this issue
* Create a new parameter `cerberus_support_contact_email` and manage the empty parameter on code : 
   * if `cerberus_support_contact_email` is null or empty, use `cerberus_support_email` parameter
   * else, use `cerberus_support_contact_email`. This solution allow the internasionalization for login page.
   * May be have you  a better idea ?

In regards to the logo, yes I will use the actual Cerberus Logo.

To finish, I will check quickly with Loic why test on Login page is KO.

Regards,

Nicolas

File: source/src/main/java/org/cerberus/database/DatabaseVersioningService.java (hunk 1 above)
Comment (vertigo17) [inline source/src/main/java/org/cerberus/database/DatabaseVersioningService.java]: I better understand and don't have any better solution ;-(.
Maybe try to get current email value by regex inside the SQL ?
I would also add just like you proposed the mention to check that parameter value is correctly after deploy that version of Cerberus.

**Comment by nicodeur** [comment]
Hi,

I update Login page with actual logo :
login-dekstop2

And I update database to delete only color yellow on`cerberus_support_email`

Travis failed on my pull request, have you an idea why ?

Regards,

Nicolas

**Comment by abourdon** [comment]
By default, Travis will try to apply the same goal for a PR as for a commit from the main branch. But Travis defines a different configuration context when integrating a commit from a PR than from the main branch.

For our case, any commit is deployed to our development server by using credentials from the main branch context. As your PR is processed with the same goal as for main branch commits, Travis tries to deploy your PR to our development server and fails because there is no credentials associated to the PR context. (Hopefully because we don't want to update our development server is a PR occurs ;-))

So, to sum up, don't care about this error for your PR, your commit compiles fine :-)

However, we have to customize Travis in order to avoid deployment for a PR.

**Comment by nicodeur** [comment]
Ok, thanks for explanation @abourdon !

**Comment by abourdon** [comment]
Evolution is done, PR could be merged now.

If I could add something else: it could be great to review the "Forgot password" page because to improve ergonomics:
- Add title and text on form to precise to user the nature of this form. E.g., : "Please enter your username. An email will be sent to your associated email address with a link to reset your password" 
- Rename "Forgot password" button to "Send me an email"
- Rename "Homepage" button by "Cancel"

What do you think @nicodeur, @vertigo17 ?

**Comment by nicodeur** [comment]
Hi,

I agree with you @abourdon,

I would like change comportment of Homepage button too. Today, it redirect (with a link) to homepage of cerberus. It would be better to manage it with javascript to show Login Page. What do you think about it ?

I can work on it next week.

**Comment by abourdon** [comment]
Sounds good to me!

Anyway, we could merge this PR as it follows same features as the current login page.

Improvement could be done in a dedicated issue and you could be assigned to it as you can now commit to the main branch.

WDYT?

**Comment by nicodeur** [comment]
Yes, it suits me perfectly !
